src/cloud-api-adaptor/pkg/securecomms/sshproxy/sshproxy.go

```diff
@@ -117,12 +119,49 @@ func (inbounds *Inbounds) AddTags(tags []string, inboundPorts map[string]string,
 	return nil
 }
 
-func (inbounds *Inbounds) Add(inPort int, name, phase string, wg *sync.WaitGroup) (string, error) {
+func (inbounds *Inbounds) listen(namespace string, tcpAddr *net.TCPAddr, name string) (tcpListener *net.TCPListener, err error) {
+	if namespace != "" {
+		var podns, myns netns.NsHandle
+		var err error
+
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+
+		if podns, err = netns.GetFromName(namespace); err != nil {
+			logger.Printf("Inbound listen get podns '%s': %v", namespace, err)
+			return nil, fmt.Errorf("inbound %s failed to get other ns '%s': %w", name, namespace, err)
+		}
+		defer podns.Close()
+		if myns, err = netns.Get(); err != nil {
+			logger.Printf("Inbound listen get myns: %v", err)
+			return nil, fmt.Errorf("inbound %s failed to get my ns '%s': %w", name, namespace, err)
+		}
+		defer myns.Close()
+
+		// switch ns
+		if err := netns.Set(podns); err != nil {
+			logger.Printf("Inbound listen setns '%s': %v", namespace, err)
+			return nil, fmt.Errorf("inbound %s failed to setns '%s': %w", name, namespace, err)
+		}
+		defer func() {
+			if err := netns.Set(myns); err != nil {
+				logger.Printf("Inbound listen revert setns: %v", err)
+			} else {
+				logger.Printf("Inbound %s listen reverted setns", name)
+			}
+		}()
+
+		logger.Printf("Inbound %s switch ns %s success", name, namespace)
+	}
+	return net.ListenTCP("tcp", tcpAddr)
```

Comment:
Could you use `netops` functions instead of directly using the `netns` package?

Something like this?
```
if namespace == "" {
    return net.ListenTCP("tcp", tcpAddr)
}

ns, err := netops.OpenNamespace(filepath.Join("/run/netns", namespace))
if err != nil {
    return ferr
}
defer ns.Close()

return ns.Run(func() error {
    return net.ListenTCP("tcp", tcpAddr)
})
```

Reply:
Thanks @yoheiueda  - have moved to `netops` functions